schemas/generate_address_space.py

```diff
@@ -3,34 +3,37 @@
 xmlparser.py is a requirement. it is in opcua folder but to avoid importing all code, developer can link xmlparser.py in current directory
 """
 import sys
+import os
+import datetime
 import logging
 # sys.path.insert(0, "..")  # load local freeopcua implementation
 #from opcua import xmlparser
 import opcua.common.xmlparser as xmlparser
 
+BASE_DIR = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
 
 def _to_val(objs, attr, val):
     from opcua import ua
     cls = getattr(ua, objs[0])
     for o in objs[1:]:
         cls = getattr(ua, _get_uatype_name(cls, o))
     if cls == ua.NodeId:
-        return "NodeId.from_string('val')"
+        return f"NodeId.from_string('{val}')"
     return ua_type_to_python(val, _get_uatype_name(cls, attr))
 
 
 def _get_uatype_name(cls, attname):
     for name, uat in cls.ua_types:
         if name == attname:
             return uat
-    raise Exception("Could not find attribute {} in obj {}".format(attname, cls))
+    raise Exception(f"Could not find attribute {attname} in obj {cls}")
```

Comment:
OK that kind of changes make that file unsuable with python 2.7 and earlier 3 version. This is fine for generate_xxx files but not for the library code